### PR TITLE
Secret rotation endpoints working

### PR DESCRIPTION
**Pull request by judy2k**
Please can we get a couple of reviews to check whether this is ready to merge?

### CODE REVIEW

File: src/HttpClient.js [hunk 1]
```diff
@@ -211,20 +211,28 @@ class HttpClient {
...
-  delete(path, callback, useJwt) {
+  delete(path, callback, useJwt, useBasicAuth) {
     let params = {};
-    if (!useJwt) {
+    if (!useJwt && !useBasicAuth) {
       params["api_key"] = this.credentials.apiKey;
       params["api_secret"] = this.credentials.apiSecret;
     }
 
+      let headers = {};
```
Comment (AlexLakatos) [inline src/HttpClient.js]: indentation looks broken here

File: src/HttpClient.js [hunk 2]
```diff
@@ -311,11 +319,16 @@ class HttpClient {
 
     path = path + joinChar + querystring.stringify(qs);
 
+      let headers = { "Content-Type": "application/json" };
```
Comment (AlexLakatos) [inline src/HttpClient.js]: indentation looks broken here

File: src/Account.js [hunk 3]
```diff
@@ -61,6 +61,22 @@ class Account {
       callback
     );
   }
+
+  listSecrets(account, callback) {
+    return this.options.api.get("/accounts/"+account+"/secrets", {}, callback, false, true);
```
Comment (AlexLakatos) [inline src/Account.js]: does this endpoint support filtering?

File: src/Account.js [hunk 4]
```diff
@@ -61,6 +61,22 @@ class Account {
...
+  }
+
+  getSecret(account, id, callback) {
```
Comment (AlexLakatos) [inline src/Account.js]: the way we handle get/list for the newer parts of the library, we combine into a single method `getSomething` that supports string or empty object params for /id or /?query=string

Comment (AlexLakatos) [reply src/Account.js]: If the endpoint doesn't support filtering, I'd still want to go with a single get method that gets passed an optional string for id and does either single item get or multiple item get

Comment (AlexLakatos) [reply src/Account.js]: the get/list combined method didn't look quite right, abandoned the idea

File: README.md [hunk 5]
```diff
@@ -348,6 +348,30 @@ For more information see https://docs.nexmo.com/tools/application-api/api-refere
 nexmo.account.checkBalance(callback);
 ```
 
+### List Account Secrets
+
+```js
+nexmo.account.listSecrets(account, callback);
```
Comment (judy2k) [inline README.md]: I think all the other libraries use the preferred parameter name `api_key` instead of `account` or `account_id`.

**Review by judy2k — COMMENTED** [review]
Looks great, but I think the (totally consistent) use of `account` breaks with our documentation policy of referring to this as `api_key`.

TBH I think account is the more accurate term, but I think for consistency's sake, this should probably be changed.

File: README.md [hunk 6]
```diff
@@ -348,6 +348,30 @@ For more information see https://docs.nexmo.com/tools/application-api/api-refere
...
+### List Account Secrets
+
+```js
+nexmo.account.listSecrets(api_key, callback);
+```
+
+### Get Account Secret
+
+```js
+nexmo.account.getSecret(api_key, secretId, callback);
```
Comment (judy2k) [inline README.md]: Now, I realise this is my fault, but I think the naming convention for this library should make it `apiKey` not `api_key`?

**Comment by judy2k** [comment]
Okay, I approve! You are a :star: !

**Pull request merged**